core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "scale_codec_impl.hpp"
+
+#include "scale/collection.hpp"
+#include "scale/compact.hpp"
+#include "scale/fixedwidth.hpp"
+
+#include "primitives/block.hpp"
+#include "primitives/scale_error.hpp"  // TODO: change path after moving scale_error to scale
+
+namespace kagome::primitives {
+
+  outcome::result<Buffer> ScaleCodecImpl::encodeBlock(
+      const Block &block) const {
+    Buffer out;
+    OUTCOME_TRY(encoded_header, encodeBlockHeader(block.header()));
+    out.putBuffer(encoded_header);
+
+    // put number of extrinsics
+    bool res = compact::encodeInteger(block.extrinsics().size(), out);
+    if (!res) {
+      std::terminate();
+      // it can't be, res is certainly true
+      // this check will disappear after refactoring scale library
+    }
+
+    for (auto &&extrinsic : block.extrinsics()) {
```

Comment:
add const

Reply:
afaik, it doesn't matter, auto can mean const auto & when necessary.